app/lib/actions/katello/repository/clone_yum_content.rb

```diff
@@ -38,6 +38,15 @@ def plan(source_repo, target_repo, filters, purge_empty_units, options = {})
             plan_copy(Pulp::Repository::CopyYumMetadataFile, source_repo, target_repo)
             plan_copy(Pulp::Repository::CopyDistribution, source_repo, target_repo)
 
+            # Check for matching content before indexing happens, the content in pulp is
+            # actually updated, but it is not reflected in the database yet.
+            output = {}
+            if target_repo.environment && !options[:force_metadata_regeneration]
```

Comment:
I believe you also need to make sure that the source_repo is not in the default content view.  I'm not seeing metadata_generate get called when i publish a new version